### DISCUSSION TITLE
Major issues with inertia

### DISCUSSION THREAD

**Discussion by ghost** [inertiajs/inertia#1907] [Ideas]
### Version:

- `@inertiajs/react` version: all

### Describe the problem:

Inertia still uses the renderToString method rather than renderToReadableStream

This means that a number of react features are unavailable. Other front end libraries also have streaming options for rendering. 

If you used the latter method, you could also have your layout file in jsx rather than php. Meaning all your front end code would be in the same language

Another major issue - inertia also places props inside html elements which makes the performance disgusting if your page contains a lot of data. 

Again, if you were using renderToReadableStream, you could use the bootstrapModules and bootstrapScriptContent methods to inject the prop data as a proper script tag

I realize this would cause breaking changes and would probably require a major version update. If you are interested in resolving these problems then I am willing to submit a PR

If this is just going to be closed as a "feature request" I will not waste my time 

### Steps to reproduce:

Install inertia

**Comment by ghost** [comment]


**Reply by ramonmalcolm10** [reply]
A issue was opened but was closed as completed, not sure what happened.  https://github.com/inertiajs/inertia/issues/1671

**Reply by derrickreimer** [reply]
Discussion around that ticket was consolidated here: https://github.com/inertiajs/inertia/discussions/1887

@kejedi your observations about the scope required to add support for Suspense seem to be in line with @nick-potts (in the aforementioned discussion). I can't speak for everyone else involved with Inertia, but having discussion around this (outside of the bug tracker) feels like the appropriate venue to kick things off, especially because the scope is large and it will require fundamental design decisions and changes.

**Comment by afgallo** [comment]
@kejedi thanks for bringing that up as I was in the process of assessing inertia as an option for our application V2 efforts. My first thought was "how aligned is inertia to React 19 and SSR in general". I guess you have answered my question; it sounds like there is significant effort going forward.

This is a shame though given the other options I am assessing such as Remix, Next.js, etc are potentially more aligned to the future of React on the client side but they don't really deliver the great Rails or Laravel experience on the backend which is very much something we want.

FYI - I have also found this video where Taylor (creator of Laravel) acknowledges inertia isn't a finished product yet and they are aware of the current challenges around maintenance: https://www.youtube.com/watch?v=NC6h1Oaz1rM&t=2085s

**Comment by nick-potts** [comment]
I looked into this for a few days, and a conclusion I had was using something like https://vike.dev/architecture could work nicely as an adapter.
